utils_cv/detection/dataset.py

```
@@ -0,0 +1,220 @@
+            image_folder: the name of the image folder
+            annotation_folder: the name of the annotation folder
+        """
+
+        self.root = Path(root)
+        self.transforms = transforms
+        self.image_folder = image_folder
+        self.annotation_folder = annotation_folder
+
+        self.ims = list(sorted(os.listdir(self.root / self.image_folder)))
+        self.annotations = list(
+            sorted(os.listdir(self.root / self.annotation_folder))
+        )
+        self.categories = self._get_categories()
+
+    def _get_categories(self) -> List[str]:
+        """ Parses all Pascal VOC formatted annoatation files to extract all
+        possible categories. """
+        categories = ["__background__"]
+        for annotation_path in self.annotations:
+            annotation_path = self.root / "annotations" / str(annotation_path)
+            tree = ET.parse(annotation_path)
+            root = tree.getroot()
+            objs = root.findall("object")
+            for obj in objs:
+                category = obj[0]
+                assert category.tag == "name"
+                categories.append(category.text)
+        return list(set(categories))
+
+    def get_image_features(
+        self, idx: int = None, rand: bool = False
+    ) -> Tuple[List[List[int]], List[str], str]:
+        """ Choose and get image from dataset.
+
+        This function returns all the data that is needed to effectively
+        visualize an image from the dataset.
+
+        Args:
+            idx: The image index to get the features of
+            rand: randomly select image
+
+        Raises:
+            Exception is idx is not None and rand is set to True
+
+        Returns:
+            A tuple of boxes, categories, image path
+        """
+        if (idx is not None) and (rand is True):
+            raise Exception("idx cannot be set if rand is set to True.")
+        if idx is None and rand is False:
```

Probably better to throw an error here that rand needs to be explicitly set to true if idx is not specified.